dev/TeachingTip/TeachingTip.h

```diff
@@ -94,21 +98,23 @@ class TeachingTip :
     void OnAutomationNameChanged(const winrt::IInspectable&, const winrt::IInspectable&);
     void OnAutomationIdChanged(const winrt::IInspectable&, const winrt::IInspectable&);
 
+    void OnContentSizeChanged(const winrt::IInspectable&, const winrt::SizeChangedEventArgs& args);
     void OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const winrt::AcceleratorKeyEventArgs& args);
     void OnCloseButtonClicked(const winrt::IInspectable&, const winrt::RoutedEventArgs&);
     void OnActionButtonClicked(const winrt::IInspectable&, const winrt::RoutedEventArgs&);
     void OnPopupOpened(const winrt::IInspectable&, const winrt::IInspectable&);
     void OnPopupClosed(const winrt::IInspectable&, const winrt::IInspectable&);
     void OnLightDismissIndicatorPopupClosed(const winrt::IInspectable&, const winrt::IInspectable&);
-    void OnTailOcclusionGridLoaded(const winrt::IInspectable&, const winrt::IInspectable&);
 
     void RaiseClosingEvent(bool attachDeferralCompletedHandler);
     void ClosePopupWithAnimationIfAvailable();
     void ClosePopup();
 
-    void SetViewportChangedEvent();
+    void SetViewportChangedEvent(const gsl::strict_not_null<winrt::FrameworkElement>& target);
     void RevokeViewportChangedEvent();
```

Comment:
what's the difference between strict_not_null and not_null? the "strict" part is an extra word that's confusing me.

Reply:
Strict_Not_Null wraps not_null and provides an explicit constructor from T.

It makes some more of the issues that would result in runtime crashes result in compile time crashes.

---
In reply to: [279092981](https://github.com/Microsoft/microsoft-ui-xaml/pull/619#discussion_r279092981) [](ancestors = 279092981)